Framework/Indexing/test/IndexInfoTest.h

```diff
@@ -109,18 +109,82 @@ class IndexInfoTest : public CxxTest::TestSuite {
     TS_ASSERT_THROWS(info.setSpectrumDefinitions(defs), std::runtime_error);
   }
 
-  void test_setSpectrumDefinitions_size_mismatch() {
+  void test_setSpectrumDefinitions_size_mismatch_cow_ptr() {
     IndexInfo info(3);
     const auto defs = Kernel::make_cow<std::vector<SpectrumDefinition>>(2);
     TS_ASSERT_THROWS(info.setSpectrumDefinitions(defs), std::runtime_error);
   }
 
-  void test_setSpectrumDefinitions() {
+  void test_setSpectrumDefinitions_cow_ptr() {
     IndexInfo info(3);
     const auto defs = Kernel::make_cow<std::vector<SpectrumDefinition>>(3);
     TS_ASSERT_THROWS_NOTHING(info.setSpectrumDefinitions(defs));
     TS_ASSERT_EQUALS(info.spectrumDefinitions().get(), defs.get());
   }
+
+  void test_StorageMode_Cloned() {
+    IndexInfo rank0(3, IndexInfo::StorageMode::Cloned,
+                    IndexInfo::Communicator{2, 0});
```

Comment:
Minor point. Might be nice to see some named arguments or inline comments for the arguments for the IndexInfo::Communicator.